client/src/app/pages/applications/analysis-wizard/analysis-wizard.tsx

```diff
@@ -51,6 +51,28 @@ interface IAnalysisWizard {
   isOpen: boolean;
 }
 
+const determineMode = (
+  applications: Application[]
+):
+  | "binary"
+  | "source-code-deps"
+  | "source-code"
+  | "binary-upload"
+  | undefined => {
+  if (applications.length === 0) return undefined;
+  const modes = applications.map((app) => {
+    const { repository, binary } = app;
+    return repository || (repository && binary)
+      ? "source-code-deps"
+      : binary && binary !== ""
+      ? "binary"
+      : undefined;
+  });
+  const firstMode = modes[0];
```

Comment:
Note that this could be reduced to  problem of removing duplicates i.e. 
 ```
// using array -> set -> array
const modes =Array.from(new Set(application.map(....
// or using radash
const modes = unique(application.map(...
```

Reply:
I fixed it and also changed the title
is it ok now